internal/tiltfile/k8s.go

```diff
@@ -281,7 +281,14 @@ func (s *tiltfileState) k8sResource(thread *starlark.Thread, fn *starlark.Builti
 	}
 
 	resourceName := workload.String()
-	manuallyGrouped := false
+	o, ok := s.k8sResourceOptions[resourceName]
```

Comment:
Here are a couple bugs to mull:
```
k8s_resource(new_name='a', objects=['config'])
# generates: Error: k8s_resource has only non-workload objects but doesn't provide a new_name
k8s_resource('a', objects=['config2'])
```
```
k8s_resource('a', objects=['config2'])
# doesn't generate an error, but the resulting resource has 'config2' and not 'config'
k8s_resource(new_name='a', objects=['config2'])
```

Relatedly, we should figure out what semantics we want when a call to `k8s_resource` sets a `new_name` - do subsequent calls to `k8s_resource` access it by its workload name or its `new_name`? Right now, `new_name` doesn't really take effect until assembly (except in the case of workload-less resources, as in the above bugs).

Reply:
Added test cases for both of those!

ya, i think this question speaks a lot to the workshop Han is working on, and making sure we have good nudges to push people in the right direction when they get the name wrong.